### ISSUE TITLE
Mutating new records in 0.16 does not include updated props or relationships in post request

### ISSUE THREAD

**Issue by chbonser** [orbitjs/ember-orbit#299]
I'm upgrading an app from 0.15.x to 0.16.

The following sequence in 0.15.x would generate a post request with attributes and relationships in the request payload. In 0.16 it only includes the object identity.

```js
let fork = this.store.fork();
let newComment = await fork.addRecord({ type: "comment" });
await newComment.update({ message: "some words", blog: blog });
await this.forkedStore.requestQueue.process();
await this.store.merge(fork);
```

In the case above, I can 'make it work', if I refactor to avoid the `update` call and set the props/relationships in the `addRecord` call. However I think the flow above is reasonable to expect to work AND my actual use cases are not always so simple.

**Comment by dgeb** [comment]
@chbonser can you please create a failing test, perhaps targeting the `release-0-16` branch? I'd like to know if this is still an issue on the `master` branch as well.

**Comment by chbonser** [comment]
Yes, working on it...

**Comment by chbonser** [comment]
So here is what I've learned...

**With Orbit 0.15.x** the above sequence generates a single `addRecord` op.

<img width="1188" alt="Screen Shot 2020-08-27 at 11 05 54 AM" src="https://user-images.githubusercontent.com/1022582/91466883-60ac7300-e855-11ea-8f5a-2495b0db068f.png">

**With Orbit 0.16.x** it generates two ops, `addRecord` and `updateRecord`. This results in a POST and a PATCH request where the POST request is missing the attributes and relationships. 

<img width="960" alt="Screen Shot 2020-08-27 at 10 58 47 AM" src="https://user-images.githubusercontent.com/1022582/91467057-a6693b80-e855-11ea-804e-17bc427e6037.png">

**Comment by dgeb** [comment]
The merging of operations is done by the [coalesceRecordOperations](https://github.com/orbitjs/orbit/blob/release-0-16/packages/%40orbit/data/src/operation.ts#L381) function. This function, as well as `recordDiffs`, have undergone some recent improvements. I'm curious if this is still an issue on `master`.

If you were to post a failing test in the orbit repo I will investigate soon.

**Comment by chbonser** [comment]
@dgeb See https://github.com/orbitjs/orbit/pull/777

**Comment by dgeb** [comment]
I believe this should now be fixed in `@orbit/data` v0.16.7, which is a dependency of just published `ember-orbit` v0.16.9.

@chbonser thanks for raising this!
